django/db/models/fields/files.py

```diff
@@ -440,8 +440,9 @@ def contribute_to_class(self, cls, name, **kwargs):
         # Attach update_dimension_fields so that dimension fields declared
         # after their corresponding image field don't stay cleared by
         # Model.__init__, see bug #11196.
-        # Only run post-initialization dimension update on non-abstract models
-        if not cls._meta.abstract:
+        # Only run post-initialization dimension update on non-abstract models,
+        # and where width_field or height_field are set.
+        if not cls._meta.abstract and (self.width_field or self.height_field):
```

Comment:
`has_dimension_fields` is no longer necessary in `update_dimension_fields`.

Reply:
This is a bit backward incompatible as `ImageField` subclasses could overwrite `update_dimension_fields()` and do more. I'd add a short note in the [backward incompatible section of release notes](https://docs.djangoproject.com/en/dev/releases/5.0/#miscellaneous).

Reply:
Thanks! I removed the redundant check and added a note in the 5.0 release notes.